protoc-gen-go/generator/name_test.go

```diff
@@ -83,3 +84,39 @@ func TestGoPackageOption(t *testing.T) {
 		}
 	}
 }
+
+func TestUnescape(t *testing.T) {
+	tests := []struct {
+		in  string
+		out string
+		err string
```

Comment:
Just check if it errored or not. String parsing on errors is generally frowned upon.